libs/execution_context/ebpf_maps.c

```diff
@@ -459,13 +466,52 @@ _create_object_hash_map(_In_ const ebpf_map_definition_in_memory_t* map_definiti
     return _create_hash_map_with_map_struct_size(sizeof(ebpf_core_object_map_t), map_definition);
 }
 
+static ebpf_core_map_t*
+_create_lru_hash_map(_In_ const ebpf_map_definition_in_memory_t* map_definition)
+{
+    ebpf_core_lru_map_t* map =
+        (ebpf_core_lru_map_t*)_create_hash_map_with_map_struct_size(sizeof(ebpf_core_lru_map_t), map_definition);
+    if (map) {
+        ebpf_result_t retval;
+        // Note:
+        // ebpf_hash_table_t doesn't require synchronization as long as allocations
+        // are performed using the epoch allocator.
+        retval = ebpf_hash_table_create(
+            &map->key_history,
+            ebpf_epoch_allocate,
+            ebpf_epoch_free,
+            map->core_map.ebpf_map_definition.key_size,
+            sizeof(uint64_t),
+            map->core_map.ebpf_map_definition.max_entries,
+            NULL);
+        if (retval != EBPF_SUCCESS) {
+            ebpf_hash_table_destroy((ebpf_hash_table_t*)map->core_map.data);
+            ebpf_free(map);
+            map = NULL;
+        }
+    }
+    if (map)
+        return &map->core_map;
+    else
+        return NULL;
```

Comment:
Could be shortened to
```suggestion
    return (map) ? &map->core_map : NULL;
```